Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/ApiGatewayTestHelper.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Text.Json;
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.TestTool.Extensions;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.UnitTests.SnapshotHelper;
+using Microsoft.AspNetCore.Http;
+using Xunit;
+
+namespace Amazon.Lambda.TestTool.UnitTests
+{
+    public class ApiGatewayTestHelper
+    {
+        private readonly SnapshotTestHelper _snapshots = new(new JsonSerializerOptions
+            {
+                WriteIndented = true,
+                Converters = { new HttpResponseMessageConverter() }
+            }
+        );
+
+        public async Task VerifyApiGatewayResponseAsync(
+            APIGatewayProxyResponse response,
+            ApiGatewayEmulatorMode emulatorMode,
+            string snapshotName)
+        {
+            // Convert response to HttpResponse (simulates what API Gateway would do)
+            var convertedResponse = await ConvertToHttpResponseAsync(response, emulatorMode);
+
+            // Load the expected response from snapshot
+            var expectedResponse = await _snapshots.LoadSnapshot<HttpResponseMessage>(snapshotName);
+
+            // Compare the responses
+            await AssertResponsesEqual(expectedResponse, convertedResponse);
+        }
+
+        public async Task VerifyHttpApiV2ResponseAsync(
+            APIGatewayHttpApiV2ProxyResponse response,
+            string snapshotName)
+        {
+            // Convert response to HttpResponse (simulates what API Gateway would do)
+            var convertedResponse = await ConvertToHttpResponseAsync(response);
+
+            // Load the expected response from snapshot
+            var expectedResponse = await _snapshots.LoadSnapshot<HttpResponseMessage>(snapshotName);
+
+            // Compare the responses
+            await AssertResponsesEqual(expectedResponse, convertedResponse);
+        }
```

Comment:
nit: these 2 functions are very close, they can be combined into a single function